common/include/pcl/pcl_macros.h

```diff
@@ -123,6 +123,7 @@
                     PCL_DEPRECATED_MODIFY_MSG(Major, Minor, Message)),  \
                 major_version_mismatch)
 
+#define PCL_DEPRECATED_HEADER(Major, Minor, Message) PCL_DEPRECATED(Major, Minor, Message)
```

Comment:
Something like this will save repetitive text like `"This header is deprecated"`.
```suggestion
#define PCL_DEPRECATED_HEADER(Major, Minor, Message) PCL_DEPRECATED(Major, Minor, "This header is deprecated. " Message)
```

We could also maybe have this if the header has been deprecated in favor of another file.
```suggestion
#define PCL_DEPRECATED_HEADER(Major, Minor, AltFile, Message) PCL_DEPRECATED_HEADER(Major, Minor, "Use " AltFile Message)
```

Reply:
If we want to use the AltFile option too, maybe we should overload the macro?

https://www.boost.org/doc/libs/1_72_0/libs/preprocessor/doc/ref/overload.html

Reply:
Do we need 2 versions? Most of the headers get deprecated in favor of another, so I don't think we'd need overloading.

Reply:
> Do we need 2 versions? Most of the headers get deprecated in favor of another, so I don't think we'd need overloading.
There might be cases where we might be removing a feature, so it  will be handy there.
Though #3761 is an exception, similar cases might come up.

Reply:
Then no need to save a few keystrokes in the message. Macros aren't supposed to be used flexibly. Overloading macros would only cause confusion (and even now, I'm thinking of how I can make the PCL_DEPRECATED macro better because the error messages aren't great)

Reply:
> Then no need to save a few keystrokes in the message. 

More than saving keystrokes, my aim was to  have a more standard and unified depreciation message for all headers.
Maybe we can have macros for standard messages defined for depreciation.

>Overloading macros would only cause confusion 

Agreed, it does add unnecessary complexity.

Reply:
Maybe we should follow similar structure as used for `PCL_DEPRECATED`:

`PCL_DEPRECATED_HEADER(1, 12, "Use new_file instead")` and use the first suggestion,
```
#define PCL_DEPRECATED_HEADER(Major, Minor, Message) PCL_DEPRECATED(Major, Minor, "This header is deprecated. " Message)
```


In case there isn't an alternative file, we can simply use:
`PCL_DEPRECATED_HEADER(1, 12, "")`, or some custom message instead of a blank.